### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by audunstrand**
Added validation for autoscaling.
@gtcno @jhrv

### CODE REVIEW

File: api/appconfig.go [hunk 1]
```diff
@@ -67,6 +68,20 @@ type ExposedResource struct {
...
+type ValidationErrors struct {
+	Errors []ValidationError
+}
+
+type ValidationError struct {
+	ErrorMessage string
+	Fields       []Field
+}
+
+type Field struct {
```
Comment (gtcno) [inline api/appconfig.go]: Do you need anything more than a map[string]string ?

File: api/appconfig.go [hunk 2]
```diff
@@ -111,5 +128,61 @@ func fetchAppConfig(url string, deploymentRequest NaisDeploymentRequest) (naisAp
...
+	validationErrors := validateAppConfig(appConfig);
+	if len(validationErrors.Errors) != 0 {
+		glog.Error("Invalid appconfig: ", validationErrors.Error())
+		return NaisAppConfig{}, &validationErrors
+	}
+
 	return appConfig, nil
 }
+
+func validateAppConfig(appConfig NaisAppConfig) ValidationErrors {
```
Comment (gtcno) [inline api/appconfig.go]: Would extracting each validation into its own function lead to better tests and readability?

File: api/appconfig_test.go [hunk 3]
```diff
@@ -83,9 +83,46 @@ func TestNoAppConfigFlagCreatesAppconfigFromDefaults(t *testing.T) {
...
+	defer gock.Off()
+	gock.New(repopath).
+		Reply(200).
+		File("testdata/nais_error.yaml")
+
+	_, err := fetchAppConfig(NaisDeploymentRequest{AppConfigUrl: repopath})
+	assert.Error(t, err)
+}
+
+func TestIncalidCouThreshold(t *testing.T) {
```
Comment (gtcno) [inline api/appconfig_test.go]: Typos

File: api/appconfig.go [hunk 4]
```diff
@@ -111,5 +128,61 @@ func fetchAppConfig(url string, deploymentRequest NaisDeploymentRequest) (naisAp
 		return NaisAppConfig{}, err
 	}
 
+	validationErrors := validateAppConfig(appConfig);
```
Comment (jhrv) [inline api/appconfig.go]: no need for ;

File: api/appconfig.go (hunk 2 above)
Comment (audunstrand) [reply api/appconfig.go]: fixing

File: api/appconfig_test.go (hunk 3 above)
Comment (audunstrand) [reply api/appconfig_test.go]: fixing

**Pull request merged**